pkg/daemon/ceph/cleanup/disk.go

```diff
@@ -179,21 +179,27 @@ func (s *DiskSanitizer) executeSanitizeCommand(osdInfo oposd.OSDInfo, wg *sync.W
 	// On return, notify the WaitGroup that we’re done
 	defer wg.Done()
 
-	output, err := s.context.Executor.ExecuteCommandWithCombinedOutput(shredUtility, s.buildShredArgs(osdInfo.BlockPath)...)
-	if err != nil {
-		logger.Errorf("failed to sanitize osd disk %q. %s. %v", osdInfo.BlockPath, output, err)
-	}
-
-	logger.Infof("%s\n", output)
-	logger.Infof("successfully sanitized osd disk %q", osdInfo.BlockPath)
+	for _, device := range []string{osdInfo.BlockPath, osdInfo.MetadataPath, osdInfo.WalPath} {
+		if device == "" {
+			continue
+		}
 
-	// If the device is encrypted let's close it after sanitizing its content
-	if osdInfo.Encrypted {
-		err := osd.CloseEncryptedDevice(s.context, osdInfo.BlockPath)
+		output, err := s.context.Executor.ExecuteCommandWithCombinedOutput(shredUtility, s.buildShredArgs(device)...)
 		if err != nil {
-			logger.Errorf("failed to close encrypted osd disk %q. %v", osdInfo.BlockPath, err)
-		} else {
-			logger.Infof("successfully closed encrypted osd disk %q", osdInfo.BlockPath)
+			logger.Errorf("failed to sanitize osd disk %q. %s. %v", device, output, err)
+		}
+
+		logger.Infof("%s\n", output)
+		logger.Infof("successfully sanitized osd disk %q", device)
```

Comment:
Looks like there was previously a bug where it printed success even after failing. So seems like a need to skip printing success if there was a failure.

Reply:
Indeed, I moved the output logging before, and then it is the same logic as for encrypted OSD output.

Reply:
Hello, @travisn, do you have any comments after the changes I made? Thanks!